src/ray/raylet/worker_pool.h

```diff
@@ -350,16 +354,19 @@ class WorkerPool : public WorkerPoolInterface, public IOWorkerPoolInterface {
   ///                             worker pool abstraction. Outside this class, workers
   ///                             will have rpc::WorkerType instead.
   /// \param job_id The ID of the job to which the started worker process belongs.
+  /// \param task_id The ID of the task which triggers the worker process starting.
   /// \param dynamic_options The dynamic options that we should add for worker command.
   /// \param serialized_runtime_env The runtime environment for the started worker
   /// process. \return The id of the process that we started if it's positive, otherwise
   /// it means we didn't start a process.
   Process StartWorkerProcess(
       const Language &language, const rpc::WorkerType worker_type, const JobID &job_id,
+      const TaskID &task_id = TaskID::Nil(),
       const std::vector<std::string> &dynamic_options = {},
       const int runtime_env_hash = 0, const std::string &serialized_runtime_env = "{}",
       std::unordered_map<std::string, std::string> override_environment_variables = {},
-      const std::string &serialized_runtime_env_context = "{}");
+      const std::string &serialized_runtime_env_context = "{}",
+      const PopWorkerCallback callback = PopWorkerCallback());
```

Comment:
I think `callback` must be non empty. So there's no need to add a default arguement.